### ISSUE TITLE
Normal Inverse Gaussian Outputs Positive log_prob

### ISSUE THREAD

**Issue by i418c** [tensorflow/probability#1782]
While training a model with a NIG output, I noticed that the loss for the negative log likelihood would go negative.  At first I thought that perhaps the values of verify in the model were outside of support for the NIG function, but fixing these values to what (I think) should be valid still would yield a negative loss.

Any help on why this occurs and how to fix it would be appreciated.

While creating an example, I noticed that the quickest way to trigger this was to set the verify values to single number.  A gist reproducing the issue can be found [here](https://gist.github.com/i418c/8e407534558ede92e39e2fbfc5e9c91f#file-normal-inverse-gaussian-negative-log_prob-ipynb).

**Comment by csuter** [comment]
This is fine. Probability densities can be large and positive, yielding
positive log prob and negative NLL loss.

**Comment by csuter** [comment]
Hehe, welcome to the wild world of high dimensional nonconvex optimization
🙂. No idea! I think most people abandon hope of provable optimality in
most cases and focus on whether a particular method solves a specific
problem satisfactorily.
